src/iptvsimple/data/Channel.cpp

```diff
@@ -499,6 +499,8 @@ bool Channel::GenerateFlussonicCatchupSource(const std::string& url)
       {
         if (fsListType == "index")
           m_catchupSource = fsHost + "/" + fsChannelId + "/timeshift_rel-{offset:1}.m3u8" + fsUrlAppend;
+        else if (fsListType == "mono")
```

Comment:
This won’t work I’m afraid as you’re forcing all `mono` list types to use your format, and for any `mono` list type they will never reach the following `else`.

The only way to do this would be to add a new catch format type so that you only test your condition and not any other `mono` types.